File: problems/1185-day-of-the-week/solution.py
class Solution:
    def dayOfTheWeek(self, day: int, month: int, year: int) -> str:
        days = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"]
        d = day
        m = month
        y = year
        if m < 3:
            m += 12
            y -= 1
        c = y // 100
        yr = y % 100
        w = (d + ((13 * (m + 1)) // 5) + yr + (yr // 4) + (c // 4) - (2 * c)) % 7
        return days[w]

File: problems/1185-day-of-the-week/solution.c
char* dayOfTheWeek(int day, int month, int year) {
    char* days[] = {"Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"};
    int t[] = {0, 3, 2, 5, 0, 3, 5, 1, 4, 6, 2, 4};
    year -= month < 3;
    return days[(year + year/4 - year/100 + year/400 + t[month-1] + day) % 7];
}